src/shared/Microsoft.Git.CredentialManager/EnvironmentBase.cs

```diff
@@ -67,5 +74,7 @@ public bool IsDirectoryOnPath(string directoryPath)
         public abstract void RemoveDirectoryFromPath(string directoryPath, EnvironmentVariableTarget target);
 
         protected abstract string[] SplitPathVariable(string value);
+
+        public abstract string LocateExecutable(string program);
```

Comment:
Here's where I think this could have an implementation and add 1 or 2 more abstracts configuring it's behavior.